src/SnapKeyring.ts

```diff
@@ -397,68 +398,173 @@ export class SnapKeyring extends EventEmitter {
     chainId?: string;
   }): Promise<Json> {
     const { account, snapId } = this.#resolveAddress(address);
-    if (!account.methods.includes(method as EthMethod)) {
-      throw new Error(
-        `Method '${method}' not supported for account ${account.address}`,
-      );
-    }
+    this.#validateMethod(method as EthMethod, account);
+
     const requestId = uuid();
 
     // Create the promise before calling the snap to prevent a race condition
     // where the snap responds before we have a chance to create it.
-    const promise = new DeferredPromise<Response>();
-    this.#requests.set(requestId, { promise, snapId });
+    const promise = this.#createPromise<Response>(requestId, snapId);
```

Comment:
`createPromise` seems a bit too generic IMO, maybe `createRequestPromise` (as we're also registering it to the request map)

Reply:
Nice point! Here you go [e29f479](https://github.com/MetaMask/eth-snap-keyring/pull/232/commits/e29f479809383fa56d6be2abb7f08c034670546a) ☺️